alf/algorithms/off_policy_algorithm.py

```diff
@@ -104,17 +104,27 @@ def _train_iter_off_policy(self):
         if not config.update_counter_every_mini_batch:
             alf.summary.get_global_counter().add_(1)
 
-        with record_time("time/unroll"):
-            training_info = self.unroll(config.unroll_length)
-            self.summarize_rollout(training_info)
-            self.summarize_metrics()
+        def _unroll():
+            with record_time("time/unroll"):
+                training_info = self.unroll(config.unroll_length)
+                self.summarize_rollout(training_info)
+                self.summarize_metrics()
+
+        if config.unroll_with_grad:
```

Comment:
use `with torch.set_grad_enabled(is_train):` to simplify code

Reply:
changed